api/XDS_PROTOCOL.md

```diff
@@ -345,6 +345,14 @@ resources to avoid resending them over the network.
 
 ![Incremental reconnect example](diagrams/incremental-reconnect.svg)
 
+#### Resource names
+Resources are identified by a resource name or an alias. Aliases of a resource, if present, can be identified by the alias field in the resource of a DeltaDiscoveryResponse. The resource name
+will be returned in the name field in the resource of a DeltaDiscoveryResponse. Envoy can send either an alias or the resource name in the resource_names_subscribe field in order to subscribe to a resource.
+
+#### Unsubscribing from Resources
+Envoy will keep track of a per resource reference count internally. This count will keep track of the total number of aliases/resource names that are currently subscribed to. When the reference count reaches zero,
```

Comment:
Maybe also point out that Envoy should use the knowledge of aliases during eviction of resources (e.g. if it's doing something like LRU) to ensure all referring internal resources are updated accordingly.

Another thing that Envoy should do; if it has a resource X and knows it has aliases Y and Z, when it comes time to query for Y, it should check its existing resources and discover X.

Reply:
Thanks updated.